build/Platform.targets

```diff
@@ -0,0 +1,156 @@
+    <Exec Command="git rev-parse HEAD 2>&amp;1" StandardOutputImportance="Low" IgnoreExitCode="true" IgnoreStandardErrorWarningFormat="true" ConsoleToMSBuild="true">
+      <Output TaskParameter="ConsoleOutput" PropertyName="LatestCommit" />
+      <Output TaskParameter="ExitCode" PropertyName="LatestCommitExitCode" />
+    </Exec>
+    <!-- We shouldn't fail the build if we can't retreive the commit hash, so in this case just set it to N/A -->
+    <PropertyGroup Condition="'$(LatestCommitExitCode)'!='0'">
+      <LatestCommit>N/A</LatestCommit>
+    </PropertyGroup>
+  </Target>
+
+  <Target Name="BuildPlatform" DependsOnTargets="GetLatestCommitHash">
+    <Error Text="Please specify the RID for the aspnetcore platform via PlatformRID: {win7-x64|win7-x86|osx-x64|linux-x64}." Condition="'$(PlatformRID)' == ''"/>
+
+    <!-- Clear working directory -->
+    <RemoveDir Directories="$(_WorkRoot)" />
+
+    <PropertyGroup>
+      <PlatformWorkDirectory>$(_WorkRoot)Platform\</PlatformWorkDirectory>
+      <PlatformOutputDirectory>$(_WorkRoot)Publish\</PlatformOutputDirectory>
+      <PlatformPublishDirectory>$(PlatformOutputDirectory)shared\Microsoft.AspNetCore.All\$(PackageVersion)\</PlatformPublishDirectory>
+
+      <_PlatformProps>Configuration=$(Configuration);RuntimeIdentifier=$(PlatformRID)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);RuntimeFrameworkVersion=$(MicrosoftNETCoreApp20PackageVersion)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);DotNetRestoreSourcePropsPath=$(GeneratedRestoreSourcesPropsPath)</_PlatformProps>
+      <_PlatformProps>$(_PlatformProps);AspNetUniverseBuildOffline=true</_PlatformProps>
+    </PropertyGroup>
+
+    <!-- Move to working dir -->
+    <ItemGroup>
+      <PlatformFiles Include="$(_TemplatesDir)Platform\**\*" />
+    </ItemGroup>
+
+    <Copy SourceFiles="@(PlatformFiles)" DestinationFolder="$(PlatformWorkDirectory)\%(RecursiveDir)" />
+    <Copy SourceFiles="$(_SrcDir)Directory.Build.props" DestinationFolder="$(_WorkRoot)" />
+
+    <!-- Set _Target=Restore so the project will be re-evaluated to include Internal.AspNetCore.Sdk MSBuild properties on the next step. -->
+    <MSBuild Projects="$(PlatformWorkDirectory)Platform.csproj"
+      Targets="Restore"
+      Properties="$(_PlatformProps);MicrosoftAspNetCoreAllVersion=$(PackageVersion);_Target=Restore" />
+
+    <!-- Publish -->
+    <MSBuild Projects="$(PlatformWorkDirectory)Platform.csproj"
+      Targets="Publish"
+      Properties="$(_PlatformProps);GenerateRuntimeConfigurationFiles=true;SelfContained=false;PublishDir=$(PlatformPublishDirectory)" />
+
+    <!-- Clean deps.json -->
+    <RepoTasks.TrimDeps DepsFiles="$(PlatformPublishDirectory)/Platform.deps.json" />
+
+    <!-- Clean up artifacts that publish generates which we don't need -->
+    <ItemGroup>
+      <ToDelete Include="$(PlatformPublishDirectory)\Platform" />
+      <ToDelete Include="$(PlatformPublishDirectory)\Platform.dll" />
```

Comment:
Have you tried disabling the compilation step of this file altogether? I'm concerned this assembly will leak into deps.json files, nuspecs, or something else.

Reply:
This is trimmed from the deps files. I'm copying this logic from https://github.com/dotnet/core-setup/blob/d5e9580461cf73577f4882ce8755a038c907abfb/src/sharedFramework/sharedFramework.proj. I'd like to keep the same logic for now, at least for the first version, and gradually improve on it later.